plugins/Marketplace/Controller.php

```diff
@@ -219,38 +221,17 @@ public function overview()
     {
         $view = $this->configureViewAndCheckPermission('@Marketplace/overview');
 
-        $show  = Common::getRequestVar('show', 'plugins', 'string');
-        $query = Common::getRequestVar('query', '', 'string');
-
-        $sort = new Sort();
-        $sort = $sort->getSort();
-
-        $mode = new Mode();
-        $mode = $mode->getMode();
-
         // we're fetching all available plugins to decide which tabs need to be shown in the UI and to know the number
         // of total available plugins
         $allPlugins = $this->plugins->getAllPlugins();
         $allThemes   = $this->plugins->getAllThemes();
         $paidPlugins = $this->plugins->getAllPaidPlugins();
 
-        $showThemes  = ($show === 'themes');
-        $showPlugins = !$showThemes;
-        $showPaid    = ($show === 'premium');
-        $showAll    = !$showPaid;
-
-        if ($showPlugins && $showPaid) {
-            $type = PurchaseType::TYPE_PAID;
-            $view->numAvailablePlugins = count($paidPlugins);
-        } elseif ($showPlugins && $showAll) {
-            $type = PurchaseType::TYPE_ALL;
-            $view->numAvailablePlugins = count($allPlugins);
-        } else {
-            $type = PurchaseType::TYPE_ALL;
-            $view->numAvailablePlugins = count($allThemes);
-        }
-
-        $pluginsToShow = $this->plugins->searchPlugins($query, $sort, $showThemes, $type);
+        $view->numAvailablePlugins = [
```

Comment:
Can we update the variable also on the back-end side?
```suggestion
        $view->numAvailablePluginsByType = [
```